controllers/nginx/pkg/cmd/controller/utils.go

```diff
@@ -39,6 +39,18 @@ func sysctlSomaxconn() int {
 	return maxConns
 }
 
+// sysctlFSFileMax returns the value of fs.file-max, i.e.
+// maximum number of open file descriptors
+func sysctlFSFileMax() int {
+	maxConns, err := sysctl.New().GetSysctl("fs/file-max")
+	if err != nil {
+		glog.Errorf("unexpected error reading system maximum number of open file descriptors (fs.file-max): %v", err)
+		return 0
+	}
+
+	return maxConns
```

Comment:
Should we have some "padding" e.g. maxConns - 1024 ?  Also, is `worker_rlimit_nofile` per worker process .  I honestly just don't know - I'm happy to follow your lead here if you say "this is correct", just not sure I follow the logic of letting a single nginx process use up all the FDs available to the whole system?